---
title: "Report of pancreas nuclei ATAC-seq (sequencing date: 05/07/2019)"
author: "SM islet group"
date: "12/07/2019"
output: 
  html_document
---
<style type="text/css">

body{ /* Normal  */
      font-size: 16px;
  }
td {  /* Table  */
  font-size: 14px;
}
h1.title {
  font-size: 28px;
}
h1 { /* Header 1 */
  font-size: 24px;
  color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 20px;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, warning = FALSE)
library(tidyverse)
```
# Results
## **Distribution of fragment length in ATAC-seq library**
We expect fragment distribution with clear nucleosome phasing. Peak at ~ 150 bp results from sequence reads that span one nucleosome, and larger peaks represent progressively multiple conpact chromatins.
<center>
![ATAC-seq library fragment length](C:/Users/lingzili/Documents/ATAC_Nuclei/graph/picard_insert_size.png){ width=50% }
</center>

## **Venn diagram of alpha-cell-selective and beta-cell-selective peaks**

## 
Islet transcription factor binding sites (FOXA2, MAFB, NKX2.2, NKX6.1, PDX1)

# Methods
## Nuclei isolation
From https://doi.org/10.1038/s41467-019-09975-4
For Islet samples
ISL_3 and ISL_4, permeabilized nuclei were obtained by resuspending cells in 250
µL Nuclear permeabilization buffer [0.2% IGEPAL-CA630 (I8896, Sigma), 1 mM
DTT (D9779, Sigma), Protease inhibitor (05056489001, Roche), 5% BSA (A7906,
Sigma) in PBS (10010-23, Thermo Fisher Scientific)], and incubating for 10 min on
a rotator at 4 °C. Nuclei were then pelleted by centrifugation for 5 min at 500 x g at
4 °C. The pellet was re-suspended in 25 µL ice-cold Tagmentation Buffer [33 mM
Tris-acetate (pH = 7.8) (BP-152, Thermo Fisher Scientific), 66 mM K-acetate
(P5708, Sigma), 11 mM Mg-acetate (M2545, Sigma), 16% DMF (DX1730, EMD
Millipore) in Molecular biology water (46000-CM, Corning)]. An aliquot was then
taken and counted by hemocytometer to determine nuclei concentration.
Approximately 50,000 nuclei were re-suspended in 20 µL ice-cold Tagmentation
Buffer, and incubated with 1 µL Tagmentation enzyme (FC-121-1030; Illumina) at
37 °C for 30 min with shaking 500 rpm. The tagmentated DNA was purified using
MinElute PCR purification kit (28004, Qiagen). The libraries were amplified using
NEBNext High-Fidelity 2x PCR Master Mix (M0541, NEB) with primer extension
at 72 °C for 5 min, denaturation at 98 °C for 30 s, followed by eight cycles of
denaturation at 98 °C for 10 s, annealing at 63 °C for 30 s and extension at 72 °C for
60 s. Amplified libraries were then purified using MinElute PCR purification kit
(28004, Qiagen), and two size selection steps were performed using SPRIselect bead
(B23317, Beckman Coulter) at 0.55X and 1.5X bead-to-sample volume rations,
respectively. For ISL_1 and ISL_2, frozen nuclear pellets of 50,000 cells were
thawed on ice, re-suspended in 50 μL of transposition reaction mix (2.5 μL of Tn5
transposase in 1x TD buffer (Illumina)) for 30 min at 37 °C in a thermomixer with
gentle shaking. Immediately following transposition, tagmented DNA was purified
using a MinElute Kit (Qiagen) or a DNA Clean and Concentrator-5 kit (Zymo) and
eluted in 10 μL of nuclease-free H2O. Five microliters of the purified sample was
PCR amplified for 12 cycles using KAPA Real-Time Library amplification kit
(KAPA Biosystems) and customized Nextera PCR primers (as in Buenrostro
et al.12). Amplified libraries were purified using AMPure XP (Beckman Coulter)
beads and eluted in 12–15 μL of nuclease-free H2O. Libraries were sequenced on
either an Illumina NextSeq 550 or Illumina HiSeq2500.

Cross-linking was carried out with 1% formaldehyde
(sigma) in PBS at room temperature (RT) for 10 min and quenched with 125 mM
Glycine (sigma) at RT for 5 min. Nuclei were isolated using a loose-fitting Dounce
homogenizer in hypotonic buffer (20 mM Hepes pH 7.9, 10 mM KCl, 1 mM
EDTA, 10% Glycerol and 1 mM DTT with additional protease inhibitor (Roche)
for 30 strokes and centrifuge at 3500 x g at 4 °C.
## FACS sort
## ATAC-seq libaray construction
## fastqc raw reads

```{bash eval=FALSE}
# Create a new directory fastqc
mkdir output_fastqc

# Create a new screen
screen -S fastqc

# Move to the directory where raw reads are stored
cd /data/home/lingzili/VDS2/NGS_Runs/190705_A00316_0085_AHCJFTDRXX/Data/Intensities/BaseCalls/Demultiplex_ATAC/KO_SM/

# fastqc raw reads
fastqc -t 16 -o /data/home/lingzili/ATAC_analysis/05072019/output_fastqc/ *.fastq.gz

# Move to the directory where fastqc output is
cd /data/home/lingzili/ATAC_analysis/05072019/output_fastqc/

# Run MultiQC to assemble all fastqc results in the current directory
multiqc .

```

Examine the fastqc results

## Reads trimming

```{bash eval=FALSE}
# Trim Nextera adapter
ls *gz | cut -d"_" -f 1 | sort -u | while read id
do
Read1=$(ls ${id}_*R1_001.fastq.gz)
Read2=$(ls ${id}_*R2_001.fastq.gz)
echo "Read1 is defined as $Read1 and Read2 is defined as $Read2"
echo "Start trimming on sample $id"
trim_galore --nextera --paired --fastqc --output_dir /data/home/lingzili/ATAC_analysis/05072019/ $Read1 $Read2
done

# Do multiqc on trimmed fastqc files
```

## Alignment with Bowtie2

```{bash eval=FALSE}
# Bowtie2 alignment
ls *gz | cut -d"_" -f 1 | sort -u | while read id
do 
Read1=$(ls ${id}_*R1_*.fq.gz)
Read2=$(ls ${id}_*R2_*.fq.gz)
echo "Read1 is defined as $Read1 and Read2 is defined as $Read2"
echo "Start alignment with sample $id"
bowtie2 -p 16 --very-sensitive -k 10 -x /data/home/lingzili/mm10_genome/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/genome -1 $Read1 -2 $Read2 -S ${id}_trim.sam
done

# Convert from sam to bam, and sort
for id in *.sam
do echo "Begin processing $id"
baseFilename=`basename $id .sam`
samtools view -@ 16 -bS $id | samtools sort -@ 16 -o ${baseFilename}.bam
echo "${id} is converted to ${baseFilename}.bam"
done

```

## Alignment QC

```{bash eval=FALSE}
# Get stat from bam file
for i in *.bam
do baseFilename=`basename $i .bam`
samtools flagstat -@ 16 $i > ${baseFilename}.stat.txt
done 

# Do multiqc on flagstat reports
multiqc .

# Size distribution of paired-end library
for j in *.bam
do echo "Analyze size distribution in $j"
echo $''
baseFilename=`basename $j .bam`
java -jar /data/home/lingzili/genomeTools/picard.jar CollectInsertSizeMetrics I=$j O=${baseFilename}.size.out H=${baseFilename}_size_histogram.pdf HISTOGRAM_WIDTH=800 INCLUDE_DUPLICATES=true M=0.1
echo "Size distribution analysis on ${baseFilename} is complete"
echo $''
done
```

## Create Bedgraph
```{bash eval=FALSE}
# MACS2 create bedGraph for visualization (paired-end, equal to HOMER create BedGraph)
for j in *.bam
do echo "Create bedGraph on $j"
baseFilename=`basename $j .bam`
macs2 pileup -i $j -o ${baseFilename}.bdg -f BAMPE
echo "${baseFilename}.bdg is done"
done

# Compress bdg file for download
for j in *.bdg
do echo "Start compression on $j"
gzip $j
done
```

## Reads clean-up
```{bash eval=FALSE}
# Remove PCR duplicated reads
samtools rmdup -s  ES_input_bestAlignment.bam ES_input_filtered.bam

# Remove mitochondrial DNA, low-quality reads (MAPQ < 30) and Encode blacklist regions
samtools view -bS -q 30 ES_input.sam > ES_input_bestAlignment.bam

```

## Peak calling and annotation
```{bash eval=FALSE}
# All mapped reads are offset by +4 bp for the +strand and -5 bp for the -strand, extended reads to 200 bp, and centered reads

# Peak calling
macs2 -q 0.05 -nomodel -shift 37 -extsize 73

macs2 --nomodel —shift -100 –extsize 200 –keep-dup all

# Differential peaks between different cell types
macs2 bdgdiff

# Join together individual peaks separated by <100 bp 

# Peaks were merged for each cell type separately using bedtools

# Read depth across peaks was normalised by applying the bamnormalise and bamsummary function of deeptools using standard settings

# Peak annotation by HOMER

# Motif analysis on peak regions by HOMER
findMotifsGenome.pl 

# Calculate read coverage for each peak in the merged set of ATAC-seq peaks, excluding peaks that overlapped blacklisted genomic regions

# Calculate the Spearman correlation between the read counts for each sample

# Promoter region (−5 kb/ + 2 kb of transcription start site) for protein-coding genes
done
```



